### PR TITLE
Make .quota.ceiling a pointer

### LINKED ISSUE #144: Ceiling not set equal zero, it should mean unlimited
In the API we mention that [no ceiling means unlimited](https://github.com/kubernetes-sigs/kueue/blob/27b05145f94ac528ff92790efc70d62d8c9feafc/api/v1alpha1/clusterqueue_types.go#L250), which is currently not true and the ceiling must be explicitly set.

We need to change the type to a pointer to distinguish between not set and 0.

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind bug
/kind api-change

#### What this PR does / why we need it:

Makes `ceiling` a pointer with the semantics that `nil` means no limit. This should simplify ClusterQueue configurations, specially when there are no cohorts.

#### Which issue(s) this PR fixes:

Fixes #144

### CODE REVIEW

File: pkg/cache/cache.go [hunk 1]
```diff
@@ -473,7 +473,9 @@ func resourceLimitsByName(in []kueue.Resource) map[corev1.ResourceName][]FlavorL
 			fLimits := FlavorLimits{
 				Name:       string(f.ResourceFlavor),
 				Guaranteed: workload.ResourceValue(r.Name, f.Quota.Guaranteed),
-				Ceiling:    workload.ResourceValue(r.Name, f.Quota.Ceiling),
+			}
+			if f.Quota.Ceiling != nil {
+				fLimits.Ceiling = pointer.Int64(workload.ResourceValue(r.Name, *f.Quota.Ceiling))
```
Comment (ahg-g) [inline pkg/cache/cache.go:478]: wouldn't it be safer if we keep flimits.Ceiling as type `int64` and just set it to MaxInt when f.Quota.Ceiling is nil?

Comment (alculquicondor) [reply pkg/cache/cache.go:478]: I think pointer is more readable. It helps with setting up unit tests too.

File: pkg/util/testing/wrappers.go [hunk 2]
```diff
@@ -315,7 +315,6 @@ func MakeFlavor(rf, guaranteed string) *FlavorWrapper {
 		ResourceFlavor: kueue.ResourceFlavorReference(rf),
 		Quota: kueue.Quota{
 			Guaranteed: resource.MustParse(guaranteed),
-			Ceiling:    resource.MustParse(guaranteed),
```
Comment (ahg-g) [inline pkg/util/testing/wrappers.go:318]: hmm, I am surprised that don't have an integration test not assuming ceiling=guaranteed. Should we add one?

File: pkg/cache/cache.go (hunk 1 above)
Comment (ahg-g) [reply pkg/cache/cache.go:478]: ok

File: pkg/util/testing/wrappers.go (hunk 2 above)
Comment (denkensk) [reply pkg/util/testing/wrappers.go:318]: IIUC, If we don't set the ceiling, the ceiling is equal to generated before. There are also some integration tests, we don't set ceiling --> assuming ceiling=guaranteed?

Comment (ahg-g) [reply pkg/util/testing/wrappers.go:318]: I don't see changes to the integration tests and they are passing, so I assumed no tests are depending on ceiling=guaranteed (meaning can't borrow)

Comment (alculquicondor) [reply pkg/util/testing/wrappers.go:318]: The CQs in our integration tests don't belong to a cohort or belong to one of size one. So the ceiling doesn't matter.

We could add a new integration test for cohort, but we can do it in a follow up.

Cohorts are tested at unit level, but it's still good to have one in integration as the multiple creation events could always lead to complications

Comment (alculquicondor) [reply pkg/util/testing/wrappers.go:318]: Updated

**Comment by ahg-g** [comment]
/hold
/lgtm

In case you want to use automated squash.

**Pull request merged**